src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/TypeMaker.java

```diff
@@ -118,7 +124,7 @@ Type makeType(jdk.internal.clang.Type t) {
         }
     }
 
-    Type makeTypeInternal(jdk.internal.clang.Type t) {
+    Type makeTypeInternal(jdk.internal.clang.Type t, List<Declaration.Variable> params) {
```

Comment:
I believe you can avoid all the changes to this method, and have a simple method on FunctionType which gets a parameter name list and gives you a _new_ function type with the list in it - e.g. `withParameterNames`. Then you can call that method from `TreeMaker` _after_ you have parsed a `Type`.